src/test/java/edu/kis/powp/jobs2d/TestCommandVisitor.java

```diff
@@ -0,0 +1,112 @@
+package edu.kis.powp.jobs2d;
+
+import edu.kis.powp.jobs2d.command.*;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Command Visitor implementation test.
+ *
+ * @author Michał Kubik & Bartosz Gralewski
+ */
+public class TestCommandVisitor {
+    private static final Job2dDriver driver = new StubDriver();
+
+    /**
+     * Command Visitor test.
+     */
+    public static void main(String[] args) {
+
+        ArrayList<DriverCommand> commandsList = new ArrayList<>(3);
+        commandsList.add(new OperateToCommand(1, 2));
+        commandsList.add(new SetPositionCommand(2, 3));
+        commandsList.add(new OperateToCommand(3, 4));
+
+        ICompoundCommand compoundCommand = new ICompoundCommand() {
+            final List<DriverCommand> driverCommands = commandsList;
+
+
+            @Override
+            public Iterator<DriverCommand> iterator() {
+                return driverCommands.iterator();
+            }
+
+            @Override
+            public void execute(Job2dDriver driver) {
+
+            }
+
+            @Override
+            public void accept(ICommandVisitor commandVisitor) {
+                for (DriverCommand elem : driverCommands) {
+                    if (elem instanceof SetPositionCommand)
+                        commandVisitor.visit((SetPositionCommand) elem);
+                    if (elem instanceof OperateToCommand)
+                        commandVisitor.visit((OperateToCommand) elem);
+                    if (elem instanceof ICompoundCommand)
+                        commandVisitor.visit((ICompoundCommand) elem);
+                }
```

Comment:
incorrect! Implementation of accept should always be the same (it can be put in the ICompoundCommand interface).